### ISSUE TITLE
GZIP JSON files fail to open

### ISSUE THREAD

**Issue by of13tch** [awslabs/amazon-s3-find-and-forget#356]
When running the forget sevice over gzip compressed JSON (with a .gz extension) we are presented with the following error:
```
"Error":"Unable to retrieve object: Not a gzipped file (b'{"')"
```
We believe the cause is because the file is opened as follows (https://github.com/XioResearchInterGalactic/amazon-s3-find-and-forget/blob/master/backend/ecs_tasks/delete_files/main.py#L157): 
```
        with s3.open_input_stream(
            "{}/{}".format(input_bucket, input_key), buffer_size=FIVE_MB
        ) as f:
```
The open_input_stream method has a default parameter `compression='detect'` which returns a gzip stream.
https://arrow.apache.org/docs/python/generated/pyarrow.fs.S3FileSystem.html#pyarrow.fs.S3FileSystem.open_input_file

After creating the stream it is initalized here https://github.com/XioResearchInterGalactic/amazon-s3-find-and-forget/blob/master/backend/ecs_tasks/delete_files/json_handler.py#L11 - the result being that the input stream is initalized again as a GZIP stream which results in the error above (not a gzipped file).

I suspect that when this was written when `compression='detect'` was not a feature.

**Comment by matteofigus** [comment]
Thanks for submitting an issue and thanks for the preliminary analysis.

I think you may be right, we recently moved from s3fs to arrow's s3 handler, and I don't see in our acceptance tests suite anything covering gzip json end-to-end, so perhaps we introduced the bug there in [v0.53](https://github.com/awslabs/amazon-s3-find-and-forget/blob/master/CHANGELOG.md#v053).

I think the next step for me is to add an end-to-end test there to reproduce the issue. If I can reproduce, then the fix may be as simple as you suggested (just remove the gzip handling as arrow should do be able to handle gzip natively).

I'll keep this thread updated.

**Comment by matteofigus** [comment]
@of13tch thanks for submitting the issue and being patient.

We were able to reproduce and have a fix as you suggested. We just released v0.58 which include a fix.
